onnxruntime/core/framework/partial_graph_execution_state.h

```diff
@@ -8,6 +8,51 @@
 #include "core/framework/execution_frame.h"
 
 namespace onnxruntime {
+
+class OrtValueCache {
+ public:
+  OrtValueCache() = default;
+
+  size_t size() const {
+    return cache_.size();
+  }
+
+  size_t count(const std::string name) const {
+    return cache_.count(name);
+  }
+
+  OrtValue at(const std::string name) const {
+    return cache_.at(name);
+  }
+
+  Status CacheOrtValue(const std::string node_arg_name, const OrtValue& value) {
+    cache_.emplace(node_arg_name, value);
+    return Status::OK();
+  }
+
+  Status GetCachedIds(std::vector<std::string>& output_keys) const {
+    output_keys.reserve(cache_.size());
+    for(auto kv : cache_) {
+      output_keys.push_back(kv.first);
+    }
+    return Status::OK();
+  }
+
+  Status DeleteOrtValue(const std::string node_arg_name) {
+    ORT_RETURN_IF(cache_.find(node_arg_name) == cache_.end(), "NodeArg not found in cache: ", node_arg_name);
+    cache_.erase(node_arg_name);
+    return Status::OK();
+  }
+
+  Status ClearCache() {
+    cache_.clear();
+    return Status::OK();
+  }
+
+ private:
+  std::unordered_map<std::string, OrtValue> cache_;
+};
```

Comment:
Can you please help me understand the rationale behind taking an unordered_map and just wrapping all of its functions with your functions?

Reply:
Initially this cache was set to hold more information, but in the current implementation, i agree that it can be avoided. Thanks for pointing it out